# Lecture 7: Introduction To Pandas

In [1]:
# Report_Card_Dictionary

report_card_dictionary = {"student_names": ["Harsha", "Ashwini", "Rohit", "Mahir"],
                          "math_scores": [50,75,65,70],
                          "chemistry_scores": [80,75,90,65],
                          "information_technology_scores": [40,45,50,65],
                          "physics_scores": [80,60,50,45]}
report_card_dictionary

{'student_names': ['Harsha', 'Ashwini', 'Rohit', 'Mahir'],
 'math_scores': [50, 75, 65, 70],
 'chemistry_scores': [80, 75, 90, 65],
 'information_technology_scores': [40, 45, 50, 65],
 'physics_scores': [80, 60, 50, 45]}

In [2]:
# Setup
import pandas as pd

In [3]:
# Converting Dictionary to DataFrame
report_card_df = pd.DataFrame(report_card_dictionary)
report_card_df

student_names  math_scores  chemistry_scores  information_technology_scores  \
0        Harsha           50                80                             40   
1       Ashwini           75                75                             45   
2         Rohit           65                90                             50   
3         Mahir           70                65                             65   

   physics_scores  
0              80  
1              60  
2              50  
3              45

In [4]:
report_card_df.head()

student_names  math_scores  chemistry_scores  information_technology_scores  \
0        Harsha           50                80                             40   
1       Ashwini           75                75                             45   
2         Rohit           65                90                             50   
3         Mahir           70                65                             65   

   physics_scores  
0              80  
1              60  
2              50  
3              45

In [5]:
# First two rows in the database.
report_card_df.head(2)

student_names  math_scores  chemistry_scores  information_technology_scores  \
0        Harsha           50                80                             40   
1       Ashwini           75                75                             45   

   physics_scores  
0              80  
1              60

In [6]:
# Sample of two rows in our data frame 
report_card_df.sample(3)

student_names  math_scores  chemistry_scores  information_technology_scores  \
2         Rohit           65                90                             50   
0        Harsha           50                80                             40   
1       Ashwini           75                75                             45   

   physics_scores  
2              50  
0              80  
1              60

In [7]:
# Create a description of our numeric columns
report_card_df.describe()

math_scores  chemistry_scores  information_technology_scores  \
count     4.000000           4.00000                       4.000000   
mean     65.000000          77.50000                      50.000000   
std      10.801234          10.40833                      10.801234   
min      50.000000          65.00000                      40.000000   
25%      61.250000          72.50000                      43.750000   
50%      67.500000          77.50000                      47.500000   
75%      71.250000          82.50000                      53.750000   
max      75.000000          90.00000                      65.000000   

       physics_scores  
count         4.00000  
mean         58.75000  
std          15.47848  
min          45.00000  
25%          48.75000  
50%          55.00000  
75%          65.00000  
max          80.00000

In [8]:
# Create a column called total score.
report_card_df["total_score"] = report_card_df["math_scores"] + report_card_df["chemistry_scores"] + report_card_df["information_technology_scores"] + report_card_df["physics_scores"]
report_card_df.head(5)


student_names  math_scores  chemistry_scores  information_technology_scores  \
0        Harsha           50                80                             40   
1       Ashwini           75                75                             45   
2         Rohit           65                90                             50   
3         Mahir           70                65                             65   

   physics_scores  total_score  
0              80          250  
1              60          255  
2              50          255  
3              45          245

#Challenge: Create a column called average score


In [9]:
report_card_df["avg_score"] = report_card_df["total_score"]/4
report_card_df

student_names  math_scores  chemistry_scores  information_technology_scores  \
0        Harsha           50                80                             40   
1       Ashwini           75                75                             45   
2         Rohit           65                90                             50   
3         Mahir           70                65                             65   

   physics_scores  total_score  avg_score  
0              80          250      62.50  
1              60          255      63.75  
2              50          255      63.75  
3              45          245      61.25

In [10]:
report_card_df.describe()

math_scores  chemistry_scores  information_technology_scores  \
count     4.000000           4.00000                       4.000000   
mean     65.000000          77.50000                      50.000000   
std      10.801234          10.40833                      10.801234   
min      50.000000          65.00000                      40.000000   
25%      61.250000          72.50000                      43.750000   
50%      67.500000          77.50000                      47.500000   
75%      71.250000          82.50000                      53.750000   
max      75.000000          90.00000                      65.000000   

       physics_scores  total_score  avg_score  
count         4.00000     4.000000   4.000000  
mean         58.75000   251.250000  62.812500  
std          15.47848     4.787136   1.196784  
min          45.00000   245.000000  61.250000  
25%          48.75000   248.750000  62.187500  
50%          55.00000   252.500000  63.125000  
75%          65.00000   255.000000  63.750000  
max          80.00000   255.000000  63.750000

# Insights
1. What is the mean chemistry score? 62.50
2. What is the lowest chemistry score? 65
3. What is the highest total score? 255
4. What is the mean total score? 251.25
5. What is the mean average score? 62.82


In [11]:
# Creating a table called report insights
report_card_insights_df = report_card_df.describe()
report_card_insights_df

math_scores  chemistry_scores  information_technology_scores  \
count     4.000000           4.00000                       4.000000   
mean     65.000000          77.50000                      50.000000   
std      10.801234          10.40833                      10.801234   
min      50.000000          65.00000                      40.000000   
25%      61.250000          72.50000                      43.750000   
50%      67.500000          77.50000                      47.500000   
75%      71.250000          82.50000                      53.750000   
max      75.000000          90.00000                      65.000000   

       physics_scores  total_score  avg_score  
count         4.00000     4.000000   4.000000  
mean         58.75000   251.250000  62.812500  
std          15.47848     4.787136   1.196784  
min          45.00000   245.000000  61.250000  
25%          48.75000   248.750000  62.187500  
50%          55.00000   252.500000  63.125000  
75%          65.00000   255.000000  63.750000  
max          80.00000   255.000000  63.750000

In [12]:
# Create a table called report called insights
report_card_insights_df = report_card_df.describe().transpose()
report_card_insights_df

count      mean        std     min       25%  \
math_scores                      4.0   65.0000  10.801234   50.00   61.2500   
chemistry_scores                 4.0   77.5000  10.408330   65.00   72.5000   
information_technology_scores    4.0   50.0000  10.801234   40.00   43.7500   
physics_scores                   4.0   58.7500  15.478480   45.00   48.7500   
total_score                      4.0  251.2500   4.787136  245.00  248.7500   
avg_score                        4.0   62.8125   1.196784   61.25   62.1875   

                                   50%     75%     max  
math_scores                     67.500   71.25   75.00  
chemistry_scores                77.500   82.50   90.00  
information_technology_scores   47.500   53.75   65.00  
physics_scores                  55.000   65.00   80.00  
total_score                    252.500  255.00  255.00  
avg_score                       63.125   63.75   63.75

In [13]:
# Create a CSV File of our Report Card Insights
report_card_insights_df.to_csv("Report Card Insights.csv")